Libraries/LibWeb/Loader/ResourceLoader.cpp

```diff
@@ -184,6 +184,85 @@ static bool should_block_request(LoadRequest const& request)
     return false;
 }
 
+struct FileLoadResult {
+    ReadonlyBytes data;
+    HTTP::HeaderMap response_headers;
+    Requests::RequestTimingInfo timing_info;
+};
+
+template<typename LoadCounterChangeCallback, typename FileHandler, typename ErrorHandler>
+static void request_file_for_load(LoadRequest& request, int& pending_loads, LoadCounterChangeCallback& on_load_counter_change, FileHandler on_file, ErrorHandler on_error)
```

Comment:
It feels a little icky the way we're passing references to `ResourceLoader`'s internals here and in subsequent commits. Could we just make this a private class function? The compiler can handle the template params fine as long as this is defined above its usage (which is the case already here).

(nit: I also find the phrasing of "request file for load" a bit awkward to read here and in the following commits)

This would look something like:
```c++
template<typename FileHandler, typename ErrorHandler>
void ResourceLoader::handle_file_load_request(LoadRequest const& request, FileHandler on_file, ErrorHandler on_error)
```